Identity.API/Controllers/AuthenticationController.cs

```diff
@@ -1,8 +1,61 @@
-﻿using Microsoft.AspNetCore.Mvc;
+﻿using Identity.BusinessLogic;
+using Identity.Core.Models;
+using Microsoft.AspNetCore.Mvc;
 
 namespace Identity.API.Controllers
 {
-    public class AuthenticationController : ControllerBase
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AuthenticationController(IAuthenticationService authenticationService, ILogger<AuthenticationController> logger) : ControllerBase
     {
+        private readonly IAuthenticationService _authenticationService = authenticationService;
+        private readonly ILogger<AuthenticationController> _logger = logger;
+
+        [HttpPost]
+        [Route("login")]
+        public async Task<IActionResult> Login([FromBody] Login login)
+        {
+            _logger.LogInformation("--- Init Login endpoint -------------");
+            if (!ModelState.IsValid)
+            {
+                _logger.LogDebug("--- Login failed. Invalid Credentials -------------");
+                return BadRequest("Invalid email or password " + ModelState);
+            }
+
+            var token = await _authenticationService.AuthenticateAsync(login);
+
+            if (token is null)
+            {
+                _logger.LogDebug("--- Login failed. User Not Found or Invalid Password -------------");
+                return BadRequest("Invalid email or password");
+            }
+
+            _logger.LogInformation("--- Login finished successfully -------------");
```

Comment:
The controllers should be as clean as possible. All logic and checks should be in the services